### PR TITLE
Account Squatting via OAuth Misconfiguration

### PR DESCRIPTION
**Pull request by RRudder**

### CODE REVIEW

File: submissions/description/server_security_misconfiguration/oauth_misconfiguration/account_squatting/template.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Account Squatting via OAuth Misconfiguration
+
+## Overview of the Vulnerability
+
+OAuth is an authorization framework used to identify and authenticate users for an application. There are a number of implementation misconfigurations which can lead to an OAuth framework being implemented insecurely.
+
+During the OAuth based account registration process, the OAuth authorization flow creates a new account for an email address and grants access via an OAuth token without checking if there is a pre-existing account for the same email address. This allows an attacker to register an account with a legitimate email and “squat” on the account, denying the legitimate user access to their account.
```
Comment (evildaemond) [inline submissions/description/server_security_misconfiguration/oauth_misconfiguration/account_squatting/template.md:7]: This is technically not correct, account squatting allows for a user to connect to the account after they have registered, for example, create an account as JoeBlogs@xyz.info then when JoeBlogs uses an OAuth login such as Google, they are logged into the same account as the original one, meaning that an attacker can get access to the account. If you need more clarification, DM me.

**Review by evildaemond — CHANGES_REQUESTED** [review]
Some technical changes needed.

File: submissions/description/server_security_misconfiguration/oauth_misconfiguration/account_squatting/template.md (hunk 1 above)
Comment (RRudder) [reply submissions/description/server_security_misconfiguration/oauth_misconfiguration/account_squatting/template.md:7]: Thanks for the pickup, will address this now

**Review by evildaemond — APPROVED** [review]
LGTM

**Pull request merged**